Source/Core/Core/NetPlayClient.h

```diff
@@ -158,6 +168,10 @@ class NetPlayClient : public TraversalClientClient
   void SendStartGamePacket();
   void SendStopGamePacket();
 
+  void SyncSaveDataResponse(bool success);
+  bool DecompressPacketIntoFile(sf::Packet& packet, const std::string& file_path);
+  bool DecompressPacketIntoBuffer(sf::Packet& packet, std::vector<u8>& out_buffer);
```

Comment:
Note: this can use the signature:

```cpp
std::optional<std::vector<u8>> DecompressPacketIntoBuffer(sf::Packet& packet, std::size_t packet_size);
```

(or, if the size will always be in the packet, the packet_size parameter can be omitted, and it can be extracted in the function itself).

which avoids an out parameter.

Reply:
How do I tell if the function failed? Return `nullptr`?

Reply:
Yeah, in the function itself, you can just do:

```cpp
return {};
```
or
```cpp
return std::nullopt;
```

for early-exit/failure cases. External to that, `std::optional` has it's own `operator bool`.

Reply:
Yeah that's what I mean. What's the different between `{}` and `std::nullopt`?

Reply:
Behaviorally, nothing.